### PR TITLE
*: Fix 'for ... in' guards

### PR DESCRIPTION
**Pull request by wking**
I'd been adding the `if (true)` “guards” because I didn't understand the ESLint warning.  After reading [the][1] [docs][2], I see that the rule is intended to protect from iterating over keys defined in prototypes. This commit uses the recommended approach from [here][2] to ensure we only iterate over directly-defined keys.

[1]: http://eslint.org/docs/rules/guard-for-in
[2]: https://javascriptweblog.wordpress.com/2011/01/04/exploring-javascript-for-in-loops/

### CODE REVIEW

**Comment by wking** [comment]
Also a tab-cleanup commit riding along.  I can split that out into a separate PR if folks want, but it seemed easier to keep the silly-cleanup stuff collected in one PR.

File: webapp/src/DepCard.js [hunk 1]
```diff
@@ -16,8 +16,9 @@ class DepCard extends PureComponent {
 
   blockerCount(nodes) {
     var count = 0;
-    for (var index in (this.props.dependencies || [])) {
-      if (true) {
+    var dependencies = this.props.dependencies || [];
+    for (var index in dependencies) {
+      if (Object.prototype.hasOwnProperty.call(dependencies, index)) {
```
Comment (VictorBjelkholm) [inline webapp/src/DepCard.js]: I looked at the link you posted now, missed that first... However, I don't think we have to be worried about Array.prototype being override, in that case, we should stop using whatever lib is doing that.

Comment (wking) [inline webapp/src/DepCard.js]: The suggested phrasing in the ESLint docs [is][1]:

    {}.hasOwnProperty.call(foo, key)

I don't see the point in instantiating a new object when we can use `Object.prototype` directly.  

And while we could probably use `dependencies.hasOwnProperty(index)` and similar without problems, both of the references I cite are guarding against a locally-overridden `hasOwnProperty`.  I'm unlikely to audit our dependencies for `hasOwnProperty` overrides ;), so erring on the side of verbose caution makes the most sense to me.


[1]: http://eslint.org/docs/rules/guard-for-in

Comment (VictorBjelkholm) [inline webapp/src/DepCard.js]: Fair enough :)

**Comment by wking** [comment]
Merged my own PR ([over a week without negative reviews][1]).

[1]: https://botbot.me/freenode/ipfs/2016-12-20/?msg=78212244&page=3

**Pull request merged**